content/curation/references/kata-ranks.md

```diff
@@ -1,105 +1,101 @@
 
-## Purple (Proficient)
+## Purple (Expert)
 
 ### 2 Kyu
 
-2 kyu kata represent a proficient programming level.
-At this level, kata require a mature understanding of complex programming concepts - concepts such as:
+2 kyu kata represent a professional programming level. Tasks at these levels start to become mini-projects, as each of the requirements for the main solution becomes challanges of their own. At this level, kata requires a mature understanding of several intricate programming concepts - concepts such as:
 
-- Complex AI/machine learning algorithms
-- Reverse engineering techniques
-- Basic interpreters and compilers
-- Basic mini-programs with multiple feature requirements (such as a basic Markdown parser)
+- Extreme-[performance](https://www.codewars.com/kata/5a5072a6145c46568800004d), [efficiency](https://www.codewars.com/kata/5908242330e4f567e90000a3), or [math](https://www.codewars.com/kata/5a331ea7ee1aae8f24000175) heavy algorithms.
+- Engineering complete [games](https://www.codewars.com/kata/5679d5a3f2272011d700000d) and [puzzles](https://www.codewars.com/kata/5b86a6d7a4dcc13cd900000b) with tricky or exacting rules and requirements. 
+- Basic [interpreters](https://www.codewars.com/kata/58e61f3d8ff24f774400002c), [parsers](https://www.codewars.com/kata/52a78825cdfc2cfc87000005), and [compilers](https://www.codewars.com/kata/597ccf7613d879c4cb00000f).
+- A mature use of [OOP](https://www.codewars.com/kata/5a27ca7ab6cfd70f9300007a) and [logic](https://www.codewars.com/kata/5ca40e41bfaf24001980d07a) to accomplish a desired goal. 
 
 ### 1 Kyu
 
-1 kyu kata represent a proficient programming level.
-At this level, kata are similar to 2 kyu but more challenging.
+1 kyu kata represents a general mastery of computer science. Tasks that achieve this level require a significant level of time and thought to complete, as the requirements are both demanding and sophisticated.
+At this level, kata requires experience, patience, and a wide range of skills ready to solve the problem.
 They may include concepts such as:
 
-- Advanced AI/machine learning algorithms
-- Complex interpreters and compilers
-- Complex Mini-programs with multiple feature requirements (such as a complete Markdown parser)
+- [Thorough performance algorithms](https://www.codewars.com/kata/5976c5a5cd933a7bbd000029) that need to quickly untangle a multitude of [large data values](https://www.codewars.com/kata/59568be9cc15b57637000054).
+- Advanced [games](https://www.codewars.com/kata/5ea6a8502186ab001427809e) and [puzzles](https://www.codewars.com/kata/5a20eeccee1aae3cbc000090) that need to be constructed in a detailed fashion.
+- Complex [interpreters](https://www.codewars.com/kata/52ffcfa4aff455b3c2000750) and [compilers](https://www.codewars.com/kata/59f9cad032b8b91e12000035).
+- Implementation of [mini-programs](https://www.codewars.com/kata/5a12755832b8b956a9000133) with [multiple feature requirements](https://www.codewars.com/kata/591f3a2a6368b6658800020).
```

Comment:
The second link leads to a 404.

Reply:
The dead link is fixed,  it now leads to full metal chemist 2. 
Should I worry about the failed checks?

Reply:
Either way, there shouldn't be any dead links now.